crates/ark/src/lsp/completions/sources/composite.rs

```diff
@@ -201,6 +201,20 @@ fn is_identifier_like(x: Node) -> bool {
         return true;
     }
 
+    // Consider when the user asks for completions with no existing
+    // text-to-complete, such as at the R prompt in the Console, in an empty R
+    // file, or on an empty line of a non-empty R file.
+    //
+    // Gesture-wise, a Positron user could do this with Ctrl + Space, which
+    // invokes the command editor.action.triggerSuggest.
+    //
+    // The nominal completion node in these cases is just the root or 'Program'
+    // node of the AST. In this case, we should just provide "all" completions,
+    // for some reasonable definition of "all".
+    if x.node_type() == NodeType::Program {
+        return true;
+    }
```

Comment:
I'm fine with the idea in general, but I don't think this belongs in `is_identifier_like()`, because the program node itself is not identifier-like.

I think at the call site it should be `is_identifier_like(node) | is_empty_line(node)` and `is_empty_line()` should contain this code.

The name `is_empty_line` is a WIP, if you have anything better to call this, feel free to use that.

Reply:
Yes, this is a definitely an improvement on the expressive-ness front. I'll do it.